File: FAB4_Coding (Omkar)/D1_LeapYearValidation.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin>>t;

    while(t--){
        string s;
        cin>>s;

        int year = stoi(s);
        
        if(year >= 1000 && year <= 9999){
            if((year % 100 != 0 && year % 4 == 0) || (year % 400 == 0)){
                cout<<"LEAP YEAR ";
            }

            else{
                cout<<"NOT LEAP YEAR ";
            }
        }

        else{
            cout<<"INVALID YEAR ";
        }
    }
}

File: FAB4_Coding(Sahil)/d1_leapYearValidation.js
'use strict';


function leapYear(lyear){
    if (lyear >= 1000 && lyear <= 9999) {
        if ((lyear % 100 != 0 && lyear % 4 == 0) || (lyear % 400 == 0)) {
            console.log(`Its a Leap year`);
        }
        else{
            console.log(`It's not a leap year`);
        }
    }
    else{
        console.log(`Invalid Input`)
    }
}

leapYear(2020)          // Its a Leap year
leapYear(27023)         // Invalid Input
leapYear(1995)          // It's not a leap year

